src/main/java/org/mskcc/cbio/oncokb/service/impl/UserDetailsServiceImpl.java

```diff
@@ -84,4 +85,17 @@ public Optional<UserDetailsDTO> findByUserIsCurrentUser() {
         return userDetailsRepository.findByUserIsCurrentUser()
             .map(userDetailsMapper::toDto);
     }
+
+    @Transactional(readOnly = true)
+    public List<UserAndUserDetailsPair> findAllUsersWithUserDetailsByUsersIn(List<User> users) {
+        List<Object[]> usersWithDetails = userDetailsRepository.findAllUsersWithUserDetailsByUsersIn(users);
+        return usersWithDetails
+            .stream()
+            .map(u -> {
+                User user = (User) u[0];
+                UserDetails userDetails = (UserDetails) u[1];
+                return new UserAndUserDetailsPair(user, userDetails);
```

Comment:
I just realize that isn't this UserDTO? Let's return the UserDTO directly?

Reply:
Yeah it is very similar. The reason is that [we add the companyDTO](https://github.com/oncokb/oncokb-public/blob/master/src/main/java/org/mskcc/cbio/oncokb/service/mapper/UserMapper.java#L40) when mapping the user to userDTO. If I used the UserDTO, then I cannot map to a CompanyDTO.

Reply:
Are you adding it here? https://github.com/oncokb/oncokb-public/pull/896/files#diff-ccdbff722436ab7c2f7d0a2d7ecc10a40b0b25de777c602ff46dce2e588d8009R41

Reply:
Yeah, I am adding the companyDTO at that line. If I returned a UserDTO, then I'm not able to add that line in the UserMapper.

Reply:
Should we add a new method in UserDTO, a copy of https://github.com/oncokb/oncokb-public/blob/master/src/main/java/org/mskcc/cbio/oncokb/service/mapper/UserMapper.java#L36 but accepting UserDetails as second parameter?